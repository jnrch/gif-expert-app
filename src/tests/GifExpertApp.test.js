import React from 'react';
import '@testing-library/jest-dom';

const { shallow } = require("enzyme")
const { GifExpertApp } = require("../GifExpertApp")

describe('Testing <GifExpertApp />', () => {

    test('should show correctly', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();        
    })

    test('should show list of categories', () => {

        const categories = ['madrid', 'murcia', 'zaragoza'];
        const wrapper = shallow(<GifExpertApp defaultCategories= { categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe( categories.length );
        
    })
    
})
