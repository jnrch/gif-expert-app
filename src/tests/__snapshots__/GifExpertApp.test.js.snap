// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Testing <GifExpertApp /> should show list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="madrid"
      key="madrid"
    />
    <GifGrid
      category="murcia"
      key="murcia"
    />
    <GifGrid
      category="zaragoza"
      key="zaragoza"
    />
  </ol>
</Fragment>
`;
